manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java

```diff
@@ -296,6 +331,17 @@ public ProcessorDTO toDTO(Processor processor) {
         dto.setCustomerId(processor.getBridge().getCustomerId());
         dto.setStatus(processor.getStatus());
         dto.setKafkaConnection(kafkaConnectionDTO);
+
+        /*
+         * If the Processor has sensitive values, fetch them from the Vault and expose them to the
+         * Shard.
+         */
+        if (processor.getVaultReference() != null) {
```

Comment:
Since the `VaultService` is implemented by AWS Secrets Manager (or more generally a Kubernetes CRD running on a remote Pod) do we need to perform the unmarshalling of the secrets here and pass to the Shard Operator or could we not reference and access the vault from the Shard direct (to avoid sending decrypted data across our HTTP connection)?